### PR TITLE
Implemented the FireResistence potion effects.

### PR DESCRIPTION
**Pull request by bibo38**
This not only provides fire resistence, but also prevents blaze fireballs.
See #2556

### CODE REVIEW

**Comment by bibo38** [comment]
```
ERROR: Newly undocumented API symbols found:

cPawn.HasEntityEffect (function)
```

It's documented in the header. But what is missing?

**Comment by bearbin** [comment]
It is exported to the plugin API. It should be documented in the APIDump plugin.

File: Server/Plugins/APIDump/APIDesc.lua [hunk 1]
```diff
@@ -1824,6 +1824,7 @@ a_Player:OpenWindow(Window);
 				GetHealth = { Return = "number" },
 				AddEntityEffect = { Params = "EffectType, {{cEntityEffect}}", Return = "", Notes = "Applies an entity effect" },
 				RemoveEntityEffect = { Params = "EffectType", Return = "", Notes = "Removes a currently applied entity effect" },
+				HasEntityEffect = { Return = "true, if the entity effect is currently applied", Params = "EffectType", Notes = "Checks, if an entity effect is currently applied" }
```
Comment (bearbin) [inline Server/Plugins/APIDump/APIDesc.lua]: Could you replace `EffectType` with `{{cEntityEffect|EffectType}}`, and do the same for the two lines above as well?

Comment (madmaxoft) [inline Server/Plugins/APIDump/APIDesc.lua]: The `Return` value should only specify the type of the return value(s). The description should be in `Notes`.
This is mainly for the ZBS IDE which can read these descriptions and then provide code completion based on them.

Comment (bibo38) [inline Server/Plugins/APIDump/APIDesc.lua]: Should be fixed now.

**Comment by madmaxoft** [comment]
Looks good to me. Thanks.

**Comment by bibo38** [comment]
Do I need to update this branch or can github automerge it?

**Comment by SphinxC0re** [comment]
Please rebase and squash the commits onto the current master

**Comment by bibo38** [comment]
Rebased it

**Pull request merged**